lib/interactivity-api/class-wp-interactivity-api-full-page-navigation.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Enables full page client-side navigation for the Interactivity API.
+ *
+ * IMPORTANT: this code is not meant to be included in Gutenberg yet. Instead,
+ * it will be part of an external plugin to test this feature separately.
+ */
+
+if ( ! class_exists( 'WP_Interactivity_API_Full_Page_Navigation' ) ) {
+
+	/**
+	 * Class to enable full page client-side navigation for the Interactivity API.
+	 */
+	class WP_Interactivity_API_Full_Page_Navigation {
+
+		private static $instance = null;
+
+		private static $unlock_message = 'I acknowledge that full-page client-side navigation is still experimental and will probably change, breaking my plugin or website on its next version.';
+
+		public function __construct() {
+			add_action( 'init', array( $this, 'set_default_mode' ), 9 );
+			add_action( 'wp_head', array( $this, 'buffer_start' ) );
+			add_action( 'wp_footer', array( $this, 'buffer_end' ) );
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_script_modules' ) );
+		}
+
+		/**
+		 * Returns whether the client navigation mode is `experimentalFullPage`.
+		 */
+		public function is_enabled() {
+			$unlock_message     = apply_filters( 'wp_interactivity_experimental_full_page_client_navigation', '' );
+			$iapi_router_config = wp_interactivity_config( 'core/router' );
+			return (
+				self::$unlock_message === $unlock_message &&
+				'experimentalFullPage' === $iapi_router_config['clientNavigationMode'] ?? ''
+			);
+		}
```

Comment:
I've added the filter to the `is_enabled` method. I don't know if this is the best approach, though. I would check if there's something similar in WordPress or Gutenberg. 🤔

Reply:
The problem is that if someone looks in the config, they will see that `clientNavigationMode` is still `experimentalFullPage`.

Perhaps it would be better to move it to `set_default_mode` and explain people that they have to add the filter in `init` with a priority of 8?